lib/activerecord-clean-db-structure/clean_dump.rb

```diff
@@ -124,7 +124,7 @@ def run
         dump.gsub!(/^CREATE( UNIQUE)? INDEX \w+ ON .+\n+/, '')
         dump.gsub!(/^-- Name: \w+; Type: INDEX\n+/, '')
         indexes.each do |table, indexes_for_table|
-          dump.gsub!(/^(CREATE TABLE #{table}\b(:?[^;\n]*\n)+\);\n)/) { $1 + "\n" + indexes_for_table }
+          dump.gsub!(/^(CREATE TABLE #{table}\b(:?[^;\n]*\n)+\);*\n(?:.*);*)/) { $1 + "\n" + indexes_for_table }
```

Comment:
Since this was already complex and is getting more complex, it would be nice if this regex could live as a variable and have a corresponding Ruby class that runs a test with the different table structure DDL dump variations you mentioned. 

For example one without a fillfactor or other table params, and one with that set. 

If you agree, maybe we could capture that in an issue for now, if you don't have time for the unit test now, and have tested this manually and it's working for you.